SignalServiceKit/src/Messages/TSMessagesManager.m

```diff
@@ -593,54 +628,65 @@ - (void)handleIncomingEnvelope:(OWSSignalServiceProtosEnvelope *)incomingEnvelop
     return [TextSecureKitEnv sharedEnv].profileManager;
 }
 
-- (void)handleIncomingEnvelope:(OWSSignalServiceProtosEnvelope *)incomingEnvelope
+- (void)handleIncomingEnvelope:(OWSSignalServiceProtosEnvelope *)envelope
                withCallMessage:(OWSSignalServiceProtosCallMessage *)callMessage
 {
-    OWSAssert(incomingEnvelope);
+    OWSAssert(envelope);
     OWSAssert(callMessage);
 
     if ([callMessage hasProfileKey]) {
         NSData *profileKey = [callMessage profileKey];
-        NSString *recipientId = incomingEnvelope.source;
+        NSString *recipientId = envelope.source;
         [self.profileManager setProfileKeyData:profileKey forRecipientId:recipientId];
     }
 
-    if (callMessage.hasOffer) {
-        [self.callMessageHandler receivedOffer:callMessage.offer fromCallerId:incomingEnvelope.source];
-    } else if (callMessage.hasAnswer) {
-        [self.callMessageHandler receivedAnswer:callMessage.answer fromCallerId:incomingEnvelope.source];
-    } else if (callMessage.iceUpdate.count > 0) {
-        for (OWSSignalServiceProtosCallMessageIceUpdate *iceUpdate in callMessage.iceUpdate) {
-            [self.callMessageHandler receivedIceUpdate:iceUpdate fromCallerId:incomingEnvelope.source];
+    // TODO: Should we do this synchronously?
```

Comment:
Significant question.  We have two options:

* If we want to dispatch these call messages to the `callMessageHandler` synchronously, we'll need to plumb the transaction through to the call service which is non-trivial. This will let us ensure that the "unprocessed message queue" managed by `OWSBatchMessageProcessor` isn't evacuated until the call message is received.
* We can do what I've done here which is a lot simpler... call messages will only be dropped if the app happens to exit before the async dispatch occurs, which is fine - if the app exits, the call will end.